src/node_dotenv.cc

```diff
@@ -13,33 +13,22 @@ using v8::String;
 
 std::vector<std::string> Dotenv::GetPathFromArgs(
     const std::vector<std::string>& args) {
-  const auto find_match = [](const std::string& arg) {
-    return arg == "--" || arg == "--env-file" || arg.starts_with("--env-file=");
-  };
   std::vector<std::string> paths;
-  auto path = std::find_if(args.begin(), args.end(), find_match);
+  for (size_t i = 1; i < args.size(); ++i) {
+    const auto& arg = args[i];
 
-  while (path != args.end()) {
-    if (*path == "--") {
-      return paths;
+    if (arg == "--" || arg[0] != '-') {
+      break;
     }
-    auto equal_char = path->find('=');
-
-    if (equal_char != std::string::npos) {
-      paths.push_back(path->substr(equal_char + 1));
-    } else {
-      auto next_path = std::next(path);
 
-      if (next_path == args.end()) {
-        return paths;
-      }
-
-      paths.push_back(*next_path);
+    if (arg.starts_with("--env-file=")) {
+      paths.push_back(arg.substr(11));  // Directly extract the path
+    } else if (arg == "--env-file" && i + 1 < args.size()) {
+      paths.push_back(args[++i]);  // Advance to the next argument
+    } else if (arg.length() == 2 && arg[1] != '-') {
+      ++i;  // Skip short argument values (like `-e <...>`)
```

Comment:
on a side note, it seems like this is an unnecessary optimization.

Reply:
It's not an optimization. If it's missing, the argument values are interpreted as filenames, which disrupts the `for` loop.

For example, the command 
```
node -e "1+1" --env-file .env
```
would be parsed as
```
executable argument filename <END>
```
instead of
```
executable argument value argument value
```

(I hope that made sense)